minimal.js

```diff
@@ -226,38 +239,68 @@ module.exports = function (dirname, keys, opts) {
   }
 
   db.addUnboxer = function (unboxer) {
-    unboxers.push(unboxer)
+    switch (typeof unboxer) {
+      case 'function':
+        unboxers.push(unboxer)
+        break
+
+      case 'object':
+        if (typeof unboxer.key !== 'function') throw new Error('invalid unboxer')
+        if (typeof unboxer.value !== 'function') throw new Error('invalid unboxer')
+        unboxers.push(unboxer)
+        break
+
+      default: throw new Error('invalid unboxer')
+    }
   }
 
   /* TODO extract to ssb-private */
   var box1 = {
-    box: function (content, recps) {
-      if (!recps.every(isFeed)) return null
+    boxer: function (content, recps, cb) {
+      if (!recps.every(isFeed)) return cb(null, null)
 
-      return ssbKeys.box(content, recps)
+      cb(null, ssbKeys.box(content, recps))
     },
-    unbox: {
-      key: function (ciphertext) { return ssbKeys.unboxKey(ciphertext, keys) },
-      value: function (ciphertext, key) { return ssbKeys.unboxBody(ciphertext, key) }
+    unboxer: {
+      key: function (ciphertext, value, cb) { 
+        if (!ciphertext.endsWith('.box')) return cb(null, null)
+        // TODO move this inside of ssb-keys
+
+        cb(null, ssbKeys.unboxKey(ciphertext, keys))
+      },
+      value: function (ciphertext, msgKey, value, cb) {
+        cb(null, ssbKeys.unboxBody(ciphertext, msgKey))
+      }
     }
   }
-  db.addBoxer(box1.box)
-  db.addUnboxer(box1.unbox)
+  db.addBoxer(box1.boxer)
+  db.addUnboxer(box1.unboxer)
   // ////////////////////////////////
 
-  // db.box = function (value, state) {
-  db.box = function (value) {
-    var recps = value.content.recps
-    if (!recps) return value.content
+  db.box = function (content, state, cb) { // state not used
+    var recps = content.recps
+    if (!recps) return cb(null, content)
 
-    if (typeof recps === 'string') recps = value.content.recps = [recps]
-    if (!Array.isArray(recps)) throw new Error('private message field "recps" expects an Array of recipients')
-    if (recps.length === 0) throw new Error('private message field "recps" requires at least one recipient')
+    if (typeof recps === 'string') recps = content.recps = [recps]
+    if (!Array.isArray(recps)) return cb(new Error('private message field "recps" expects an Array of recipients'))
+    if (recps.length === 0) return cb(new Error('private message field "recps" requires at least one recipient'))
```

Comment:
Do you think it would be wise to move the `if (!recps.every(isFeed)) return cb(null, null)` code here? Seems like that would be valid for any boxer, not just the default `box1` boxer, but I'm not confident about that.

Reply:
No this is going to be different with different boxers. The new private groups for example is gonna let you do

```
  recps: [ @<base64 feedId>.sha256, %<base64 groupId>.cloaked ]
```